quic: flaky TestResourceManagerAcceptDenied

```
  === RUN   TestResourceManagerAcceptDenied
      conn_test.go:49: using a RSA key: QmcUskC9aKvLxhAW6gVo1fModsSqMVFYRZM2mAjG4i9wqm
      conn_test.go:49: using a Ed25519 key: 12D3KooWBEXjh4fSvPcpaDDb3aktDsK8V3HDTB3zX88GzqGtRcsF
      transport.go:220: Unexpected call to *network.MockConnManagementScope.Done([]) at /Users/runner/work/go-libp2p/go-libp2p/p2p/transport/quic/transport.go:220 because: there are no expected calls of the method "Done" for that receiver
  --- FAIL: TestResourceManagerAcceptDenied (0.30s)

--- a/p2p/transport/quic/conn_test.go
+++ b/p2p/transport/quic/conn_test.go
@@ -198,9 +198,13 @@ func TestResourceManagerAcceptDenied(t *testing.T) {
 	clientRcmgr.EXPECT().OpenConnection(network.DirOutbound, false).Return(clientConnScope, nil)
 	clientConnScope.EXPECT().SetPeer(serverID)
 	conn, err := clientTransport.Dial(context.Background(), ln.Multiaddr(), serverID)
-	require.NoError(t, err)
-	_, err = conn.AcceptStream()
-	require.Error(t, err)
+	// In rare instances, the connection gating error will already occur on Dial.
+	if err != nil {
+		clientConnScope.EXPECT().Done()
+	} else {
+		_, err = conn.AcceptStream()
+		require.Error(t, err)
+	}
 	select {
 	case <-connChan:
 		t.Fatal("didn't expect to accept a connection")

